src/pages/library/DataBrowserPreview.js

```diff
@@ -52,7 +52,20 @@ const DataBrowserPreview = ({ id }) => {
   const [viewJSON, setViewJSON] = useState()
 
   const tableMap = _.keyBy('name', tables)
-  const tableNames = _.map('name', tables)
+  const selectOptions = _.flow([
+    _.partition(table => table.rowCount > 0),
+    _.toPairs,
+    _.flatMap(([rowIndex, tables]) => {
+      const sortedTables = _.flow([
+        _.sortBy('name'),
+        _.map(({ name, rowCount }) => { return { value: name, rowCount } })
+      ])(tables)
+
+      return rowIndex === '0' ?
```

Comment:
Is `rowIndex` a string?

Reply:
it ISSS! I was trying to debug why that was so weird, but the _.toPairs returns them as a string. It was really confusing! Is there a better solution?

Reply:
It may be possible to simplify but it's a bit difficult to make a suggestion without knowing the shapes of input/outputs. Would you be able to write a unit test for the `flow` content, which would document the input/output assumptions and we could iterate from there?

Reply:
Check out `Utils.toIndexPairs`

Reply:
alright, I added a unit test that shows how just toPairs works. The return value for the index is a string. This was the way it was recommended to me that we handle array indexing, since fp lodash map method works differently from the normal map `_.map(['a', 'b'], (value, index) => { return [index, value] })`

I would much rather prefer to be able to just grab the index from the function itself, but I don't know what the proper way of doing this with fp is. I've tried using the fp-convert method, but was told that toPairs is the style for this repo

Reply:
Oh wow, didnt notice the new comment, haha. Updated to use Utils.toIndexPairs

Reply:
Sorry, my unit testing comment earlier was not clear. I meant to encourage you to write unit tests for the entire `flow` of functions, which is:

```javascript
      const getSelectOptions = _.flow(
        _.partition(table => table.rowCount > 0),
        Utils.toIndexPairs,
        _.flatMap(([rowIndex, tables]) => {
          const sortedTables = _.flow(
            _.sortBy('name'),
            _.map(({ name, rowCount }) => ({ value: name, rowCount }))
          )(tables)

          return rowIndex ?
            { label: 'Tables without data', options: sortedTables } :
            sortedTables
        })
      )
```

This helps readers of the code understand input/outputs, and helps try out various combinations of combinators (e.g. `_.toPairs`, `Utils.toIndexPairs`).

Not a blocker for this PR, though.